pkg/intent/intent.go

```diff
@@ -115,11 +116,26 @@ func (i *Store) SetPod(node string, manifest pods.PodManifest) (time.Duration, e
 	if err != nil {
 		return 0, err
 	}
+
+	podService := &consulapi.AgentServiceRegistration{
+		Name: manifest.ID(),
+	}
+	if manifest.StatusPort != 0 {
+		podService.Check = &consulapi.AgentServiceCheck{
+			// prints the HTTP response on stderr, while exiting 0 if the status code is 200
+			Script:   fmt.Sprintf(`if [[ $(curl 0.0.0.0:%v/_status -s -o /dev/stderr -w "%%{http_code}") == "200" ]] ; then exit 0 ; else exit 2; fi`, manifest.StatusPort),
```

Comment:
I see one example in consul docs where they are referring to localhost in a
service check: http://www.consul.io/intro/getting-started/checks.html

But I see nothing about who's actually running the check. I'm going to
spelunk in consul's code a bit, I don't think there's any documentation for
this.

On Mon, Jan 12, 2015 at 4:00 PM, petertseng <notifications@github.com>
wrote:

> In pkg/intent/intent.go
> <https://github.com/square/p2/pull/53#discussion-diff-22833893>:
>
> > @@ -115,11 +116,26 @@ func (i *Store) SetPod(node string, manifest pods.PodManifest) (time.Duration, e
> >  	if err != nil {
> >  		return 0, err
> >  	}
> > +
> > +	podService := &consulapi.AgentServiceRegistration{
> > +		Name: manifest.ID(),
> > +	}
> > +	if manifest.StatusPort != 0 {
> > +		podService.Check = &consulapi.AgentServiceCheck{
> > +			// prints the HTTP response on stderr, while exiting 0 if the status code is 200
> > +			Script:   fmt.Sprintf(`if [[ $(curl 0.0.0.0:%v/_status -s -o /dev/stderr -w "%%{http_code}") == "200" ]] ; then exit 0 ; else exit 2; fi`, manifest.StatusPort),
>
> no vip (wouldn't check the right host). interesting question of what goes
> here
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/p2/pull/53/files#r22833893>.
>



-- 
Stephen